### PR TITLE
Add Search Clear Event

### PR DESCRIPTION
**Pull request by nmanu1**
Add event type for SEARCH_CLEAR_BUTTON and export AnalyticsService.

J=SLAP-1820
TEST=manual

Check that event fires correctly from test-site.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -18,4 +18,5 @@ export function provideAnalytics(config: AnalyticsConfig): AnalyticsReporter {
   return new AnalyticsReporter(config, httpRequester);
 }
 
-export * from './models';
\ No newline at end of file
+export * from './models';
+export { AnalyticsReporter };
```
Comment (yen-tt) [inline src/index.ts:22]: do we need to export AnalyticsReporter if user can invoke provideAnalytics?

Comment (nmanu1) [reply src/index.ts:22]: yes, we need it to add typing when defining context for it and for what's returned by a `useAnalytics` hook (see use in the component library [PR](https://github.com/yext/answers-react-components/pull/17))

Comment (oshi97) [reply src/index.ts:22]: Does exporting AnalyticsService work instead?

Comment (oshi97) [reply src/index.ts:22]: it might even make sense for provideAnalytics to have return type `AnalyticsService` instead

Comment (oshi97) [reply src/index.ts:22]: if I'm being perfectly honest I'm not sure what the `AnalyticsService` interface existence gets us in the first place besides letting us rename `AnalyticsReporter` and add arbitrary non-private properties, but this is probably just fussing over minutiae

Comment (nmanu1) [reply src/index.ts:22]: yeah, `AnalyticsService` works. I can use that instead for `provideAnalytics`

**Review by oshi97 — APPROVED** [review]

**Pull request merged**